examples/advanced/rag/embedding/README.md

```diff
@@ -0,0 +1,94 @@
+
+## Centralized Training
+### Single-session training
+Centralized trainings, as the baseline for comparison with FL results, are done with the following command:
+```
+bash train_single_session.sh
+```
+
+### Adaptation Step 1: iterative training
+To adapt the centralized training script to federated application, under `launch_once = true` setting, we first need to "break" the single call to `trainer.train()` into iterative calls, one for each round of training.
+For this purpose, we provided `utils/train_iterative.py` as an example, which is a modified version of `utils/train_single_session.py`.
+
+In the iterative training script, the `trainer.train()` call is replaced by a `for` loop, and the training epochs are split into six rounds, `unit_train_epochs = 0.25` epoch per round, in total `0.25 * 6 = 1.5` epochs, same as single session setting. 
+
+The first round is trained with `trainer.train()`, then from the second round, 
+we call `trainer.train(resume_from_checkpoint=True)` with `args.num_train_epochs` incremented by `unit_train_epochs` to continue training from the last checkpoint.
+
+To run iterative training, we use the following command:
+``` 
+bash train_iterative.sh
+```
+
+The training loss curves are shown below, single session and iterative scripts align with each other. 
+
+![iter_single](./figs/iter_single.png)
+
+### Adaptation Step 2: federated with NVFlare
+Once we have the iterative training script ready with "starting model" loading capability, it can be easily adapted to a NVFlare trainer by using [Client API](../../../hello-world/ml-to-fl/pt/README.md).
+
+The major code modifications are for receiving the global model, set it as the starting point for each round's training, and returning the trained model after each local training round.
+
+## Job for NVFlare FL Training
+With the local training script ready, we can go ahead to generate the NVFlare job configs by reusing the job templates.
+
+Let's set the job template path with the following command.
+```bash
+nvflare config -jt ./job_template/
+```
+Then we can check the available templates with the following command.
+```bash
+nvflare job list_templates
+```
+We can see the "sag_pt_deploy_map" template is available, with which we further generate job configs for embedding model training as:
+```
+nvflare job create -force \
+  -j "/tmp/embed/nvflare/job" -w "sag_pt_deploy_map" -sd "code" \
+  -f meta.conf min_clients=3 \
+  -f app_1/config_fed_client.conf app_script="train_fl.py" app_config="--dataset_name nli" \
+  -f app_2/config_fed_client.conf app_script="train_fl.py" app_config="--dataset_name squad" \
+  -f app_3/config_fed_client.conf app_script="train_fl.py" app_config="--dataset_name quora" \
+  -f app_server/config_fed_server.conf model_class_path="st_model.SenTransModel" components[0].args.model.args.model_name="microsoft/mpnet-base" min_clients=3 num_rounds=7 key_metric="eval_loss" negate_key_metric=True 
+```
```

Comment:
Is it possible to use new Job API for this?

Reply:
Thanks! working on it now

Reply:
done and updated